docs/FAQ.md

```diff
@@ -0,0 +1,236 @@
+
+### … handle GraphQL errors?
+
+Each genqlient-generated helper function returns two results, a pointer to a response-struct, and an error.  The response-struct will always be initialized (never nil), even on error.  If the request returns a valid GraphQL response containing errors, the returned error will be [`As`-able](https://pkg.go.dev/errors#As) as [`gqlerror.List`](https://pkg.go.dev/github.com/vektah/gqlparser/v2/gqlerror#List), and the struct may be partly-populated (if one field failed but another was computed successfully).  If the request fails entirely, the error will be another error (e.g. a [`*url.Error`](https://pkg.go.dev/net/url#Error)), and the response will be blank (but still non-nil).
+
+For example, you might do one of the following:
+```go
+// return both error and field:
+resp, err := GetUser(...)
+return resp.User.Name, err
+
+// handle different errors differently:
+resp, err := GetUser(...)
+var errList *gqlerror.List
+if errors.As(err, &errList) {
+  for _, err := range errList {
+    fmt.Printf("%v at %v\n", err.Message, err.Path)
+  }
+  fmt.Printf("partial response: %v\n", resp)
+} else if err != nil {
+  fmt.Printf("http/network error: %v\n", err)
+} else {
+  fmt.Printf("successful response: %v\n", resp)
+}
+```
+
+### … require 32-bit integers?
+
+The GraphQL spec officially defines the `Int` type to be a [signed 32-bit integer](https://spec.graphql.org/draft/#sec-Int).  GraphQL clients and servers vary wildly in their enforcement of this; for example:
+- [Apollo Server](https://github.com/apollographql/apollo-server/) explicitly checks that integers are at most 32 bits
+- [gqlgen](https://github.com/99designs/gqlgen) by default allows any integer that fits in `int` (i.e. 64 bits on most platforms)
+- [Apollo Client](https://github.com/apollographql/apollo-client) doesn't check (but implicitly is limited to 53 bits by JavaScript)
+- [shurcooL/graphql](https://github.com/shurcooL/graphql) requires integers be passed as a `graphql.Int`, defined to be an `int32`
+
+By default, genqlient maps GraphQL `Int`s to Go's `int`, meaning that on 64 bit systems there's no client-side restriction.  If you prefer to limit integers to `int32`, you can set a binding in your `genqlient.yaml`:
+
+```yaml
+bindings:
+  Int:
+    type: int32
+```
+
+Or, you can bind it to any other type, perhaps one with size-checked constructors; see the [`genqlient.yaml` documentation](`genqlient.yaml`) for more details.
+
+## How do I make a query with …
+
+### … a specific name for a field?
+
+genqlient supports GraphQL field-aliases, and uses them to determine the Go struct field name.  For example, if you do
+```graphql
+query MyQuery {
+  myGreatName: myField
+}
+```
+and genqlient will generate a Go field `MyGreatName string`.  Note that the alias will always be uppercased, to ensure the field is visible to the Go JSON library.
```

Comment:
This is assuming that `myField` was already a string, right?

Reply:
Ah, yeah, I'll add some notes to that effect.  Or maybe just call it `myString`!